### PR TITLE
183383678 Search for ping things

### PR DESCRIPTION
**Pull request by maciejkocylapc**
#### What's this PR do?
- filter ping thing table based on the minimum time 

#### How should this be manually tested?
- run dev/create_test_ping_things.rb to populate ping things
- run tests
- check http://localhost:3000/ping-thing and see if the filter works as expected

#### What are the relevant tickets?
- [URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/183383678)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

File: app/javascript/packs/ping_thing/ping_thing_table.vue [hunk 1]
```diff
@@ -1,72 +1,116 @@
...
+            <tr v-for="(pt) in all_or_filtered" :key="pt.id">
+              <td class="text-nowrap">
+                <span v-html="success_icon(pt.success)"></span>
+                <strong class="text-success h6">{{ pt.response_time.toLocaleString() }}</strong>&nbsp;ms
+              </td>
+              <td class="small">
+                {{ formatted_date(pt.reported_at) }}<br />
+                <span class="word-break">
+                  <a :href="link_from_signature(pt.signature)" target="_blank" class="small">
+                    {{ pt.signature.substring(0,6) + "..." + pt.signature.substring(pt.signature.length - 4, pt.signature.length) }}
```
Comment (kamtarnowski) [inline app/javascript/packs/ping_thing/ping_thing_table.vue:53]: Please create a method containing that logic.

File: app/javascript/packs/ping_thing/ping_thing_table.vue [hunk 2]
```diff
@@ -1,72 +1,116 @@
...
+    data() {
+      return {
+        filter_time: null,
+        api_url: '/api/v1/ping-thing/' + this.network,
+        ping_things_filtered: []
+      }
+    },
+    computed: {
+      all_or_filtered() {
+        if(this.ping_things_filtered.length > 0){
```
Comment (kamtarnowski) [inline app/javascript/packs/ping_thing/ping_thing_table.vue:108]: If there are none filtered records user will get all of the records instead. Isn't this misleading?

File: app/javascript/packs/ping_thing/ping_thing_table.vue [hunk 3]
```diff
@@ -99,6 +143,19 @@
         } else {
           return " - "
         }
+      },
+      get_filtered_records() {
+        var ctx = this
+
+        axios.get(ctx.api_url + "?time_filter=" + this.filter_time)
```
Comment (kamtarnowski) [inline app/javascript/packs/ping_thing/ping_thing_table.vue:150]: Please add additional params using the object:
```suggestion
        axios.get(
          ctx.api_url,
          { params: { time_filter: this.filter_time }}
        ).then(...)
```

File: app/javascript/packs/ping_thing/ping_thing_table.vue [hunk 4]
```diff
@@ -99,6 +143,19 @@
...
+           .then(function(response) {
+             ctx.ping_things_filtered = response.data;
+             console.log(ctx.ping_things_filtered)
```
Comment (kamtarnowski) [inline app/javascript/packs/ping_thing/ping_thing_table.vue:153]: Remove pls.

**Review by kamtarnowski — CHANGES_REQUESTED** [review]

**Comment by kamtarnowski** [comment]
As I mentioned above, there is a problem if the filtered records array is blank. See a below:
screenshot-localhost_3000-2022 10 20-11_07_07
screenshot-localhost_3000-2022 10 20-11_07_38

File: app/javascript/packs/ping_thing/ping_thing_table.vue [hunk 5]
```diff
@@ -147,14 +151,15 @@
...
-        axios.get(ctx.api_url + "?time_filter=" + this.filter_time)
-           .then(function(response) {
-             ctx.ping_things_filtered = response.data;
-             console.log(ctx.ping_things_filtered)
-           })
+        axios.get(ctx.api_url, { params: { time_filter: ctx.filter_time }})
+             .then(function(response) {
+               ctx.ping_things_filtered = response.data;
+               ctx.show_filtered_records = true
+             })
```
Comment (kamtarnowski) [inline app/javascript/packs/ping_thing/ping_thing_table.vue:158]: Fix the indentation.

**Comment by kamtarnowski** [comment]
screenshot-localhost_3000-2022 10 20-13_34_30

Could user receive in this scenario any message like "0 records found" or "No Ping Things found"?

One more - could you add the @keyup.enter to the input? https://vuejs.org/guide/essentials/event-handling.html#key-modifiers

File: app/javascript/packs/ping_thing/ping_thing_table.vue (hunk 5 above)
Comment (maciejkocylapc) [reply app/javascript/packs/ping_thing/ping_thing_table.vue:158]: it's indented in line with .get

**Review by kamtarnowski — APPROVED** [review]

**Review by kbiala — APPROVED** [review]

**Pull request merged**